9.tcga-classify/scripts/nbconverted/visualize-results.r

```diff
@@ -15,53 +15,54 @@ full_mutation_df <- load_results(results_path = mut_path, process_output = TRUE)
 full_cancertype_df <- load_results(results_path = cancertype_path, process_output = TRUE)
 
-# Setup plotting logic for supplmental plot
-genes <- unique(full_mutation_df$gene_or_cancertype)
-
-top_25 <- full_mutation_df %>%
-  dplyr::filter(gene_or_cancertype %in% genes[1:25])
-bottom_25 <- full_mutation_df %>%
-  dplyr::filter(gene_or_cancertype %in% genes[26:length(genes)])
+# Setup plotting logic for supplmental plots
+genes <- unique(levels(full_mutation_df$gene_or_cancertype))
+cancertypes <- as.character(unique(full_cancertype_df$gene_or_cancertype))
 
 # Create mutation plots
-top25_gg <- plot_mutation_figure(df = top_25)
-bottom25_gg <- plot_mutation_figure(df = bottom_25)
+gg_list <- list()
+for (plot_idx in seq(1, 50, 10)) {
+    end_idx <- plot_idx+9
+    use_genes <- genes[plot_idx:end_idx]
+    
+    subset_df <- full_mutation_df %>%
+        dplyr::filter(gene_or_cancertype %in% use_genes)
+    
+    gg_list[[use_genes[1]]] <- plot_mutation_figure(df = subset_df)
+}
 
 # Create cancertype plots
-cancertype_gg <- plot_mutation_figure(df = full_cancertype_df)
-
-sup_legend_gg <- cowplot::get_legend(cancertype_gg) 
-
-# Plot on cowplot grid
-supplement_plot <- (
-  cowplot::plot_grid(
-    top25_gg + theme(legend.position = "none"),
-    bottom25_gg + theme(legend.position = "none"),
-    cancertype_gg + theme(legend.position = "none"),
-    labels = c("A", "", "B"),
-    ncol = 1,
-    nrow = 3
-  )
-)
-
-supplement_plot <- cowplot::plot_grid(
-    sup_legend_gg,
-    supplement_plot,
-    rel_heights = c(0.04, 1),
-    ncol = 1,
-    nrow = 2
-)
-
-supplement_plot
+for (plot_idx in seq(1, 33, 11)) {
+    end_idx <- plot_idx+10
+    use_cancertypes <- cancertypes[plot_idx:end_idx]
+    
+    subset_df <- full_cancertype_df %>%
+        dplyr::filter(gene_or_cancertype %in% use_cancertypes)
+    subset_df$gene_or_cancertype <- as.character(subset_df$gene_or_cancertype)
+    
+    gg_list[[use_cancertypes[1]]] <- plot_mutation_figure(df = subset_df)
+}
 
-# Save Figure
-for (extension in c(".png", ".pdf")) {
-  fig_file <- paste0("supplementary_figure_tcga_classify_auroc", extension)
-  fig_file <- file.path("figures", fig_file)
-  cowplot::save_plot(filename = fig_file,
-                     plot = supplement_plot,
-                     base_height = 12.5,
-                     base_width = 18)
+# Save a series of plots
+for (g_idx in 1:length(names(gg_list))) {
+    g_name <- names(gg_list)[g_idx]
+    g <- gg_list[[g_name]] + theme(legend.position = "bottom")
+
+    # Save Figure
+    for (extension in c(".png", ".pdf")) {
+      fig_file <- paste0("supplementary_figure_tcga_classify_auroc_plotindex_",
+                         g_idx,
+                         "_name_",
+                         g_name,
+                         extension)
+
+      fig_file <- file.path("figures", fig_file)
+      ggplot2::ggsave(filename = fig_file,
+                      plot = g,
+                      height = 150,
```

Comment:
You won't need to change this height and width in the future?

Reply:
unlikely... well, maybe. I will add variables